sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala

```diff
@@ -49,7 +49,7 @@ class ConsoleSink(options: Map[String, String]) extends Sink with Logging {
     println("-------------------------------------------")
     // scalastyle:off println
     data.sparkSession.createDataFrame(
-      data.sparkSession.sparkContext.parallelize(data.collect()), data.schema)
```

Comment:
I don't think this means we can't do anything.  I just think that we need to fix the query plan and call take without changing the plan.  Its kind of a hack but it would work until we make the planner smarter.

I think something like `data.queryExecution.executedPlan.executeTake(...)` would be safe.